### ISSUE TITLE
Squiz.Classes.SelfMemberReference: false negative if file contains namespace operator

### ISSUE THREAD

**Issue by jrfnl** [PHPCSStandards/PHP_CodeSniffer#553]
## Describe the bug

The `namespace` keyword can both be used for namespace declarations as well as as an operator - "magic keyword" - to resolve to the current namespace.
See: https://www.php.net/manual/en/language.namespaces.nsconstants.php#example-298

This last case is not correctly taken into account when determining the current namespace, which leads to false negatives.

### Code sample
```php
<?php

namespace TestMe;

//namespace\functionCall();

class SelfMemberReference
{
    public function falseNegative()
    {
        $testResults[] = \TestMe\SelfMemberReference::test();
    }
}
```

### To reproduce
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs -ps ./test.php --standard=squiz --sniffs=squiz.classes.selfmemberreference`
3. See error message displayed
```text
------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------
 11 | ERROR | [x] Must use "self::" for local static member reference (Squiz.Classes.SelfMemberReference.NotUsed)
------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
------------------------------------------------------------------------------------------------------------------
```
4. Now uncomment line 5 containing the `namespace` keyword used as an operator.
5. Run `phpcs -ps ./test.php --standard=squiz --sniffs=squiz.classes.selfmemberreference` again.
6. See no error displayed **<= this is the false negative.**

## Expected behavior

That `namespace` keywords used as an operator are ignored when determining the applicable namespace name.

## Versions (please complete the following information)

|                                       |                                                                                                           |
|--------------------------|----------------------------------------------------------------------  |
| Operating System           | _not relevant_  (Windows 10)                                                   |
| PHP version                    | _not relevant_  (8.3.8)                                                                             |
| PHP_CodeSniffer version | 3.10.1 / `master` as of today                                                                        |
| Standard                         | Squiz                                                      |
| Install type                      | _not relevant_ (git clone) |

## Please confirm

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.
